### PR TITLE
DROID-167: MeatNet add new node specific messages

### PR DESCRIPTION
**Pull request by angrygorilla**
- Added Read Firmware Rev, Hardware Rev, and Session Info Message req/resp
- Added MeatNet Model Info req/resp definitions; formatting changes
- Added MeatNet Log req/resp; added trim to Hardware Revision string

### CODE REVIEW

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionRequest.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+ package inc.combustion.framework.ble.uart.meatnet
+ import inc.combustion.framework.ble.putLittleEndianUInt32At
+
+
+internal class NodeReadFirmwareRevisionRequest(
+    serialNumber: UInt
```
Comment (miwright2) [inline combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionRequest.kt:34]: Can we have this pass in as a string, and convert it to UInt internally?  We use `serialNumber: String` as the handle for a probe throughout the app.

Comment (miwright2) [reply combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionRequest.kt:34]: nm, moving this feedback.

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionRequest.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    NodeMessageType.PROBE_FIRMWARE_REVISION
+) {
+
+    companion object {
+        const val PAYLOAD_LENGTH: UByte = 4u
+
+        /**
+         * Helper function that builds up payload of request.
+         */
+        fun populatePayload(
```
Comment (miwright2) [inline combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionRequest.kt:46]: I think it is this function that should take in serial number as a string.

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionResponse.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package inc.combustion.framework.ble.uart.meatnet
+
+import inc.combustion.framework.ble.getLittleEndianUInt32At
+
+internal class NodeReadFirmwareRevisionResponse (
+    val serialNumber: UInt,
```
Comment (miwright2) [inline combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionResponse.kt:34]: Same, this number converted to a string -- and formated.  There is likely another example in of this when we handle advertising data.

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadHardwareRevisionRequest.kt [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        const val PAYLOAD_LENGTH: UByte = 4u
+
+        /**
+         * Helper function that builds up payload of request.
+         */
+        fun populatePayload(
+            serialNumber: UInt
+        ) : UByteArray {
+
+            val payload = UByteArray(PAYLOAD_LENGTH.toInt())
```
Comment (miwright2) [inline combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadHardwareRevisionRequest.kt:50]: `UByteArray(PAYLOAD_LENGTH.toInt()) { 0u }` will zero initialize that buffer.

**Review by miwright2 — APPROVED** [review]
Looks good.  A couple of small comments on the interface and initializing buffers to be zero-filled.

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionRequest.kt (hunk 1 above)
Comment (angrygorilla) [reply combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionRequest.kt]: updated

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionResponse.kt (hunk 3 above)
Comment (angrygorilla) [reply combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadFirmwareRevisionResponse.kt]: updated

File: combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadHardwareRevisionRequest.kt (hunk 4 above)
Comment (angrygorilla) [reply combustion-android-ble/src/main/java/inc/combustion/framework/ble/uart/meatnet/NodeReadHardwareRevisionRequest.kt:50]: updated all payload requests

**Pull request merged**